### PR TITLE
Prompt for reason for skipping.

### PR DESCRIPTION
**Pull request by othrayte**
This is a draft for #310 (Add ability to tag tasks as unanswerable).

It tasks onboard the feedback that we want to guide people away from skipping a bit and that we want to know why they have skipped. The leading idea for that appeared to be allowing the user to provide a custom reason, so that is the direction this went.

The main reason that this is only a draft PR is that at the moment we are not able to NACK after ACK-ing and the website auto-ACKs on fetching tasks so there is no chance to NACK. In this PR I have commented out the auto-ACK-ing to allow for testing this.

### CODE REVIEW

File: website/src/pages/api/new_task/[task_type].ts [hunk 1]
```diff
@@ -37,7 +37,8 @@ const handler = async (req, res) => {
   });
 
   // Update the backend with our Task ID
-  await oasstApiClient.ackTask(task.id, registeredTask.id);
```
Comment (fozziethebeat) [inline website/src/pages/api/new_task/[task_type].ts:40]: Controversial proposal: What if we really delete this here and move the auto-ack to when we submit the answer?

File: website/src/pages/api/reject_task.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Prisma } from "@prisma/client";
+import { getToken } from "next-auth/jwt";
+import { oasstApiClient } from "src/lib/oasst_api_client";
+
+/**
+ * Returns a new task created from the Task Backend.  We do a few things here:
```
Comment (fozziethebeat) [inline website/src/pages/api/reject_task.ts:6]: You'll want to delete these comments

**Review by fozziethebeat — COMMENTED** [review]
Overall I think this makes sense if we move the auto-acking.

File: website/src/pages/api/new_task/[task_type].ts (hunk 1 above)
Comment (AbdBarho) [reply website/src/pages/api/new_task/[task_type].ts:40]: in that case, shouldn't the auto ack happen in the (real) backend?

I don't see the advantage of acking & submitting immediately.

TBH, I still don't fully understand why we ack in the first place.

Comment (fozziethebeat) [reply website/src/pages/api/new_task/[task_type].ts:40]: I agree the acking is a bit weird.  Let's discuss in the next call about how needed it really is.  For now let's pretend its required and just work around the API as is.

**Comment by othrayte** [comment]
I've moved the ACK to just before submitting interactions. Also I've fixed the summerisation and labeling tasks temporarily by making them skip without NACK-ing as they don't yet use the new Task.tsx component and I wasn't sure if someone was already working on doing that and figured that would be the better way to handle them.

**Comment by othrayte** [comment]
Apologies for any review request spam, I'm still learning how that action works on github.

**Comment by fozziethebeat** [comment]
Looks good, we can fix any small issues later.

**Review by fozziethebeat — APPROVED** [review]

**Pull request merged**